src/graphnet/utils/cluster/condor.py

```diff
@@ -0,0 +1,843 @@
+        with open(self.logFilePath) as logFile:
+
+            # Get log file as text (is not too big)
+            logFileText = logFile.read()
+
+            # Get time stamp string from file text
+            timeStampLength = 14
+            pos = logFileText.find(key)
+            if pos > -1:  # check found key
+                if pos > timeStampLength:
+                    timeStamp = logFileText[pos - timeStampLength : pos]
+
+                    # Time stamp from condor job log files is annoyingly missing the year, guess it from the file modification date
+                    # print time.gmtime(os.path.getmtime(self.logFilePath))
+                    # logFileModTime = time.strftime('%m/%d/%Y', time.gmtime(os.path.getmtime(self.logFilePath)) )
+                    logFileModTime = datetime.datetime.fromtimestamp(
+                        os.path.getmtime(self.logFilePath)
+                    )
+                    timeStamp = str(logFileModTime.year) + "/" + timeStamp
+
+                    # Now parse time stamp
+                    parsedTime = datetime.datetime.strptime(
+                        timeStamp, "%Y/%m/%d %H:%M:%S "
+                    )
+
+                    return parsedTime
+
+        # If here, didn't manage to parse
+        return None
+
+    def getReturnCodeFromFile(self, logFileText):
+
+        # Parse return code from file
+        returnCodeLength = 1
+        key = "return value "
+        keyPos = logFileText.find(key)
+        if keyPos > -1:  # check found key
+            returnCodePos = keyPos + len(key)
+            if len(logFileText) > returnCodePos + 1:
+
+                # Grad the return code, handling -ve values
+                returnCode = logFileText[
+                    returnCodePos : returnCodePos + returnCodeLength
+                ]
+                if returnCode == "-":
+                    returnCode += logFileText[
+                        returnCodePos
+                        + returnCodeLength : returnCodePos
+                        + returnCodeLength
+                        + 1
+                    ]
+                return int(returnCode)
+
+        # If here, didn't manage to parse
+        return None
+
+    def getMemoryUsageFromFile(self, logFileText):
+
+        # Parse memory usage from file
+        # Break into lines, find line with table containing memory info, then tokenise the oine and get the cells containing the memory used and requested
+        key = "Memory (MB)"
+        logFileLines = logFileText.split("\n")
+        for line in logFileLines:
+            if key in line:
+                tokens = line.split()
+                if len(tokens) == 6:
+                    memoryUsed = int(tokens[3])
+                    memoryRequested = int(tokens[4])
+                    memoryAllocated = int(tokens[5])
+                    return memoryUsed, memoryRequested, memoryAllocated
+
+        # If here, didn't manage to parse
+        return None, None, None
```

Comment:
I don't see this class referenced elsewhere (apart from one or more places that are not themselves references elsewhere)?
```suggestion
```